bindinfo/handle.go

```diff
@@ -838,57 +837,111 @@ func GenerateBindSQL(ctx context.Context, stmtNode ast.StmtNode, planHint string
 			return ""
 		}
 	}
-	// We need to evolve plan based on the current sql, not the original sql which may have different parameters.
-	// So here we would remove the hint and inject the current best plan hint.
-	hint.BindHint(stmtNode, &hint.HintsSet{})
-	bindSQL := utilparser.RestoreWithDefaultDB(stmtNode, defaultDB, "")
```

Comment:
It seems we should remove the origin hints first. But we don't do it in the new implementation.

Reply:
hint.BindHint is set directly instead of append,thus we not need remove origin hints first